### PR TITLE
Update docs and add editorial board page

### LINKED ISSUE #100: Document the Editorial Board in this repo
[PEP 732 – The Python Documentation Editorial Board](https://peps.python.org/pep-0732/) was accepted last month, which formed a Documentation Editorial Board to replace the old Documentation Working Group.

Let's update this repo's docs to reflect this.

* [x] https://github.com/python/docs-community/blob/main/README.md -> PR https://github.com/python/docs-community/pull/99
* [x] https://docs-community.readthedocs.io/en/latest/workgroup/index.html

Please could the EB take care of updating https://docs-community.readthedocs.io/en/latest/workgroup/index.html?

There's lots of stuff there about (old WG) practices which the new board is best placed to update. I expect a lot can be deleted and/or replaced with a link to the PEP.

### PR DESCRIPTION
**Pull request by willingc**
This PR:
- adds an editorial board page
- orphans old pages
- edits pages to bring more up to date
- adds intersphinx targets
- updates readme

Closes #100 

----
📚 Documentation preview 📚: https://docs-community--103.org.readthedocs.build/en/103/

### CODE REVIEW

File: docs/community/community-guide.md [hunk 1]
```diff
@@ -8,20 +8,25 @@ team!
 
 There are a few resources that are particularly useful for team members. Here's
 a quick list to get you started.
-- devguide
-- docs
+
+- [devguide](https://devguide.python.org)
+- [docs](https://docs.python.org)
 - python_docs_theme
 - discourse channel
```
Comment (nedbat) [inline docs/community/community-guide.md:15]:
```suggestion
- [discourse category](https://discuss.python.org/c/documentation/26)
```

File: docs/community/community-guide.md [hunk 2]
```diff
@@ -8,20 +8,25 @@ team!
...
 ### General information about communication channels
 
 We are trying to organize our discussions in order to help both contributors and
 maintainers find and choose the right communication channels and have a positive experience :-)
 
 In this respect, we are using:
-1. [BPO](https://bugs.python.org) and GitHub issues for specific discussions related to changing a repository's content. This is determined by doc project.
+
+1. GitHub issues for specific discussions related to changing a repository's content.
 2. The [Discourse forum](http://discuss.python.org/) for general discussions, support
```
Comment (nedbat) [inline docs/community/community-guide.md:26]:
```suggestion
2. The [Discourse forum](https://discuss.python.org/c/documentation/26) for general discussions, support
```

File: docs/community/community-guide.md (hunk 1 above)
Comment (hugovk) [inline docs/community/community-guide.md:15]:
```suggestion
- [Python Docs Sphinx Theme](https://github.com/python/python-docs-theme)
- [Discourse category](https://discuss.python.org/c/documentation/26)
```

File: docs/community/community-guide.md [hunk 3]
```diff
@@ -8,20 +8,25 @@ team!
...
 a quick list to get you started.
-- devguide
-- docs
+
+- [devguide](https://devguide.python.org)
+- [docs](https://docs.python.org)
 - python_docs_theme
 - discourse channel
-
+- discord channel
```
Comment (hugovk) [inline docs/community/community-guide.md:16]:
```suggestion
- Discord server
```

**Comment by willingc** [comment]
Thanks @hugovk and @nedbat for the review. I incorporated all of your changes. It is green so I'm going to 🚢

**Pull request merged**